src/main/java/datastorage/CaregiverDAO.java

```diff
@@ -0,0 +1,98 @@
+package datastorage;
+
+import model.Caregiver;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+/**
+ * Implements the Interface <code>DAOImp</code>. Overrides methods to generate specific caregiver-SQL-queries.
+ */
+public class CaregiverDAO extends DAOimp<Caregiver> {
+
+    /**
+     * constructs Object. Calls the Constructor from <code>DAOImp</code> to store the connection.
+     * @param conn
+     */
+    public CaregiverDAO(Connection conn) {
+        super(conn);
+    }
+
+    /**
+     * generates a <code>INSERT INTO</code>-Statement for a given caregiver
+     * @param caregiver for which a specific INSERT INTO is to be created
+     * @return <code>String</code> with the generated SQL.
+     */
+    @Override
+    protected String getCreateStatementString(Caregiver caregiver) {
+        return String.format("INSERT INTO caregiver (firstname, surname, telephone) VALUES ('%s', '%s', '%s')",caregiver.getFirstName(), caregiver.getSurname(), caregiver.getTelephone());
+    }
+
+    /**
+     * generates a <code>select</code>-Statement for a given key
+     * @param key for which a specific SELECT is to be created
+     * @return <code>String</code> with the generated SQL.
+     */
+    @Override
+    protected String getReadByIDStatementString(long key) {
+        return String.format("SELECT * FROM caregiver WHERE cid = %d", key);
+    }
+
+    /**
+     * maps a <code>ResultSet</code> to a <code>Caregiver</code>
+     * @param result ResultSet with a single row. Columns will be mapped to a caregiver-object.
+     * @return caregiver with the data from the resultSet.
+     */
+    @Override
+    protected Caregiver getInstanceFromResultSet(ResultSet result) throws SQLException {
+        Caregiver c = null;
+        c = new Caregiver(result.getInt(1), result.getString(2), result.getString(3), result.getString(4));
+        return c;
```

Comment:
```suggestion
        return new Caregiver(result.getInt(1), result.getString(2), result.getString(3), result.getString(4));
```
Unnötige memory allocation und es ist schöner zu lesen